src/ConnectedMode/BranchMatcher.cs

```diff
@@ -35,10 +35,11 @@ public interface IBranchMatcher
     {
         /// <summary>
         /// Calculates the Sonar server branch that is the closest match to the head branch.
-        /// Defaults to branch marked as "main" when no match is found or when there are no head branches.
+        /// Returns null if there are no head branches
```

Comment:
Does this mean that there are no branches at all, or just that the repo is currently on a headless commit?